From ce3a5eccf7703023467691f14a7ad5d7043766af Mon Sep 17 00:00:00 2001
From: Liwei Song <liwei.song@windriver.com>
Date: Thu, 2 Jan 2020 17:38:13 +0800
Subject: [PATCH] system/mount.te: allow mount on symlink file

allow mount on symlink file /mnt

AVC avc:  denied  { read } for  pid=644 comm="mount" name="mnt"
dev="sda3" ino=865 scontext=root:sysadm_r:mount_t:s0-s15:c0.c1023
tcontext=system_u:object_r:default_t:s0 tclass=lnk_file permissive=0

Upstream-Status: Pending

Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 policy/modules/system/mount.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/system/mount.te b/policy/modules/system/mount.te
index 0b426b30e..422d411e3 100644
--- a/policy/modules/system/mount.te
+++ b/policy/modules/system/mount.te
@@ -98,6 +98,9 @@ files_list_all_mountpoints(mount_t)
 files_dontaudit_write_all_mountpoints(mount_t)
 files_dontaudit_setattr_all_mountpoints(mount_t)
 
+files_read_all_symlinks(mount_t)
+files_getattr_all_symlinks(mount_t)
+
 fs_getattr_xattr_fs(mount_t)
 fs_getattr_tmpfs(mount_t)
 fs_getattr_rpc_pipefs(mount_t)
-- 
2.17.1

